vendor/sigs.k8s.io/kustomize/kyaml/openapi/kubernetesapi/v1184/swagger.go

```diff
@@ -0,0 +1,251 @@
+// Copyright 2019 The Kubernetes Authors.
```

Comment:
~90% of the added lines (450k/490k) are these inlined swagger.json schemas:

1.18.4 / 1.18.6 / 1.18.8 appear to have no differences:
```diff
$ diff vendor/sigs.k8s.io/kustomize/kyaml/openapi/kubernetesapi/{v1184,v1186}/swagger.json
16332c16332
<     "version": "v1.18.4"
---
>     "version": "v1.18.6"

$ diff vendor/sigs.k8s.io/kustomize/kyaml/openapi/kubernetesapi/{v1184,v1188}/swagger.json
16332c16332
<     "version": "v1.18.4"
---
>     "version": "v1.18.8"
```

1.19.0 / 1.19.1 appear to have no differences:
```diff
$ diff vendor/sigs.k8s.io/kustomize/kyaml/openapi/kubernetesapi/{v1190,v1191}/swagger.json
17080c17080
<     "version": "v1.19.0"
---
>     "version": "v1.19.1"
```

It seems strange to ship a 1.21 kubectl with a 1.18 and 1.19 schema compiled into it. Some questions about those compiled-in schemas:
* Is there a reason those schema versions are inlined?
* Is some functionality of kustomize limited to those versions?
* What would happen if those schemas were omitted?
* What do those add to the binary size and startup time?

Reply:
Thanks, great questions.

- we need one relatively recent version to get data like merge keys  - in general trying to do less special case type-related coding, and switch over to configuring from openapi data.  That's a work in progress.  We don't need two.  We can drop one (again, it's a work in progress).

- we could switch to a higher version number.  yes it seems weird to have the skew, but in practice it doesn't really matter to people slinging Deployments and ConfigMaps.  it will matter more as time passes to get newer types.  Not a big deal to change (though we should do so as a later PR, which just touches a one or two go.mod files rather than 500 files).

- omitting the schemes would require some other means of obtaining merge keys and other benefits of type data - i.e. hardcoding, heuristics etc. what kustomize has done historically.

- stock gcloud linux amd64 kubectl is ~56M (v1.16).  My local kubectl build with this PR is ~42M (double checked).  Regardless, building kustomize by itself without the openapi data means ~10M change in binary size.  Yes, it takes up space.  The alternative is downloading and loading the data separately as a config file.

-  startup time to `kubectl version` between the two kubectls just mentioned is in the sub 10ms noise. 

I think we need one openapi version builtin and the change in size is the cost of doing so.   The kustomize functionality should work without talking to an api server.

Reply:
> we need one relatively recent version to get data like merge keys  - in general trying to do less special case type-related coding, and switch over to configuring from openapi data.  That's a work in progress.  We don't need two.  We can drop one (again, it's a work in progress).

ok. a follow-up to trim from 5 inlined schemas to 1 would be good.

> we could switch to a higher version number. yes it seems weird to have the skew, but in practice it doesn't really matter to people slinging Deployments and ConfigMaps. it will matter more as time passes to get newer types

I couldn't tell what happens if the user tries to make use of new types or new fields not present in the compiled-in schema.  Does kustomize attempt to get the current schema from the server, or does it reject the user's manifest with an "unknown type" or "unknown field" error, or does it have a sensible default fallback behavior?

> The kustomize functionality should work without talking to an api server.

I could see having one inlined as a fallback if the schema for the current server could not be obtained (or if there's no current server connection info), but if the current server's schema is available, shouldn't it be used? 

> startup time to kubectl version between the two kubectls just mentioned is in the sub 10ms noise.

ok, that's good.